src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -531,32 +531,25 @@ public String getContentDisplay()
             String tmp = content;
             for (User user : getMentionedUsers())
             {
-                if (isFromType(ChannelType.PRIVATE) || isFromType(ChannelType.GROUP))
-                {
-                    tmp = tmp.replace("<@" + user.getId() + '>', '@' + user.getName())
-                            .replace("<@!" + user.getId() + '>', '@' + user.getName());
-                }
+                String name;
+                if (isFromType(ChannelType.TEXT) && getGuild().isMember(user))
+                    name = getGuild().getMember(user).getEffectiveName();
                 else
-                {
-                    String name;
-                    if (getGuild().isMember(user))
-                        name = getGuild().getMember(user).getEffectiveName();
-                    else name = user.getName();
-                    tmp = tmp.replace("<@" + user.getId() + '>', '@' + name)
-                            .replace("<@!" + user.getId() + '>', '@' + name);
-                }
+                    name = user.getName();
+                tmp = tmp.replace("<@" + user.getId() + '>', '@' + name)
+                         .replace("<@!" + user.getId() + '>', '@' + name);
```

Comment:
Lets replace this double call to `replace` with a single call to `replaceAll` using a regex that covers both a nickname and non-nickname mention (make the `!` optional)